doc/source/ray-core/doc_code/pattern_map_reduce.py

```diff
@@ -0,0 +1,34 @@
+import ray
+
+# __single_threaded_map_reduce_start__
+items = list(range(100))
+map_outputs = [i * 2 for i in items]
+reduce_output = sum(map_outputs)
+assert reduce_output == 9900
+# __single_threaded_map_reduce_end__
+
+
+# __parallel_map_reduce_start__
+@ray.remote
+def map(func, obj):
+    return func(obj)
+
+
+@ray.remote
+def reduce(func, *objs):
+    return func(objs)
+
+
+map_outputs = [map.remote(lambda i: i * 2, i) for i in items]
+
+# simple reduce
+reduce_output = ray.get(reduce.remote(sum, *map_outputs))
```

Comment:
We should explain the * syntax and how it relates to Ray - t's a common mistake for people to pass the list of ObjectRefs as a single arg instead of unpacking.